File: prototipo_html/patch-ocupacao.js
const fs = require('fs');

let f2 = fs.readFileSync('foco-02.js', 'utf8');

const targetStr = "// Limpa o interval pois os dados já foram carregados";

const ocupacaoLogic = `
            // 3. OCUPAÇÃO
            const mockSituacaoOcupacional = getFieldFromLowestRip('situacao_ocupacional');
            const mockTempoDesocupacao = getFieldFromLowestRip('tempo_desocupacao');
            const mockDataOcupacao = getFieldFromLowestRip('data_ocupacao') || getFieldFromLowestRip('data_conhecimento_ocupacao');
            const mockObsOcupado = getFieldFromLowestRip('obs_ocupado') || getFieldFromLowestRip('observacoes_ocupacao');

            const radiosSituacao = document.querySelectorAll('input[name="situacao_ocupacional"]');
            
            if (radiosSituacao.length > 0) {
                // Bloqueia as opções porque a origem é a API
                radiosSituacao.forEach(r => {
                    r.disabled = true;
                    if (mockSituacaoOcupacional && r.value.toLowerCase() === mockSituacaoOcupacional.trim().toLowerCase()) {
                        r.checked = true;
                    }
                });

                if (typeof toggleOcupacao === 'function') {
                    toggleOcupacao(); // Dispara a exibição dos blocos corretos
                } else if (window.toggleOcupacao) {
                    window.toggleOcupacao();
                }

                // Elementos dos sub-blocos
                const inputTempoDesoc = document.getElementById('tempo_desocupacao');
                const inputDataOcup = document.getElementById('data_ocupacao');
                const inputObsOcup = document.getElementById('obs_ocupado');

                const formGrp = radiosSituacao[0].closest('.form-group');
                const labelOcupacao = formGrp ? formGrp.querySelector('label') : null;

                if (!mockSituacaoOcupacional) {
                    // Sem informação de ocupação na base: aplica hint ao grupo
                    if (labelOcupacao) {
                        labelOcupacao.style.backgroundColor = '#fff3cd';
                        labelOcupacao.style.padding = '4px 8px';
                        labelOcupacao.style.borderRadius = '4px';
                        labelOcupacao.style.display = 'inline-block';
                        if (!labelOcupacao.querySelector('.hint-icon')) {
                            labelOcupacao.innerHTML += ' <span class="hint-icon" title="Dado ausente na base oficial. Se tiver o dado e desejar atualizá-lo, use o campo de Observações." style="cursor:help;">ℹ️</span>';
                        }
                    }
                    
                    // Bloqueia e aplica hint aos campos filhos (mesmo que escondidos)
                    setHint(inputTempoDesoc, '');
                    setHint(inputDataOcup, '');
                    setHint(inputObsOcup, '');
                } else {
                    // Tem situação ocupacional. Aplica hint nos sub-campos de acordo com a seleção
                    const blocoDesocupado = document.getElementById('bloco-desocupado');
                    const blocoOcupado = document.getElementById('bloco-ocupado');

                    // Note: setHint expects empty value to trigger yellow hint
                    if (blocoDesocupado && blocoDesocupado.style.display !== 'none') {
                        setHint(inputTempoDesoc, mockTempoDesocupacao);
                    }
                    if (blocoOcupado && blocoOcupado.style.display !== 'none') {
                        setHint(inputDataOcup, mockDataOcupacao);
                        setHint(inputObsOcup, mockObsOcupado);
                    }
                }
            }

            // Limpa o interval pois os dados já foram carregados`;

if (f2.includes('// 3. OCUPAÇÃO')) {
    console.log('Already patched.');
} else {
    f2 = f2.replace(targetStr, ocupacaoLogic);
    fs.writeFileSync('foco-02.js', f2);
    console.log('Ocupacao patched.');
}

File: prototipo_html/scripts/patch-ocupacao.py
import re

with open('foco-02.js', 'r', encoding='utf-8') as f:
    content = f.read()

target = '''        // Radios: situacao_ocupacional
        if (dados.situacao_ocupacional) {
            const radios = bloco.querySelectorAll(`input[name="imoveis[${index}][situacao_ocupacional]"]`);
            radios.forEach(r => { if (r.value === dados.situacao_ocupacional) r.checked = true; });
        }'''

replacement = '''        // Radios: situacao_ocupacional com regras de exibição
        const ocupRadios = bloco.querySelectorAll(`input[name="imoveis[${index}][situacao_ocupacional]"]`);
        const blocoDesocupado = document.getElementById(`bloco-desocupado_${rip}`);
        const blocoOcupado = document.getElementById(`bloco-ocupado_${rip}`);
        const blocoUsoAtual = document.getElementById(`bloco-uso-atual_${rip}`);
        
        function updateOcupacao() {
            let val = '';
            ocupRadios.forEach(r => { if (r.checked) val = r.value; });
            if (val === 'Desocupado') {
                if (blocoDesocupado) blocoDesocupado.style.display = 'flex';
                if (blocoOcupado) blocoOcupado.style.display = 'none';
                if (blocoUsoAtual) blocoUsoAtual.style.display = 'none';
            } else if (val === 'Ocupado regularmente' || val === 'Ocupado irregularmente') {
                if (blocoDesocupado) blocoDesocupado.style.display = 'none';
                if (blocoOcupado) blocoOcupado.style.display = 'flex';
                if (blocoUsoAtual) blocoUsoAtual.style.display = 'flex';
            } else {
                if (blocoDesocupado) blocoDesocupado.style.display = 'none';
                if (blocoOcupado) blocoOcupado.style.display = 'none';
                if (blocoUsoAtual) blocoUsoAtual.style.display = 'none';
            }
        }
        
        ocupRadios.forEach(r => {
            r.addEventListener('change', updateOcupacao);
        });
        
        if (dados.situacao_ocupacional) {
            ocupRadios.forEach(r => { if (r.value === dados.situacao_ocupacional) r.checked = true; });
        }
        updateOcupacao();'''

if target in content:
    content = content.replace(target, replacement)
    with open('foco-02.js', 'w', encoding='utf-8') as f:
        f.write(content)
    print("foco-02.js patched with ocupacao rules!")
else:
    print("Could not find target block in foco-02.js")
